pkg/scheduler/scheduler.go

```diff
@@ -0,0 +1,104 @@
+package scheduler
+
+import (
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/the-maldridge/nbuild/pkg/graph"
+	"github.com/the-maldridge/nbuild/pkg/types"
+)
+
+func NewScheduler(l hclog.Logger, c CapacityProvider, url string) *Scheduler {
+	x := Scheduler{
+		l:                l.Named("scheduler"),
+		capacityProvider: c,
+		apiClient:        graph.NewAPIClient(l),
+		queue:            make([]Build, 0),
```

Comment:
If you're explicitly creating this with no capacity, there's no need for this `make()` you can just omit it from the instatiation of this variable.